docs/DeveloperGuide.md

```diff
@@ -1,38 +1,360 @@
+Pros: Avoid extra step of typing another command to understand a feature well.
+Cons: Wordy and long manual page when command is man, user has to scroll through
+a long manual page to find what they need. Long method.
+
+### 3.2. Parser
+
+The `Parser` is a class that handles the commands keyed in by the user.  
+It has multiple checks to ensure that the user keys in valid commands. 
+
+<!---elaborate more on the class, still unsure of the format...--->
+<!---insert other commands here, still unsure of the format...(should it all be under Parser class or separate?)--->
+
+### 3.3. Add Modules
+
+`add` command:
+
+The `add` command is used by the user to add a module or multiple modules at once.
+
+The sequence by which the Parser class handles the `add` command is as follows:
+1) The `Parser` class extracts the necessary fields from the user input, and calls upon the `addModule`
+   method in the `ModuleList` class in a loop.
+2) The method then checks the type of module that is being added, then calls the relevant
+   constructor (`Core`, `UE`, `GE`, `Internship`).
+3) The `addModule` method returns the module that is added to the `Parser` class.
+4) The `getModuleListSize` method is called from `ModuleList` class.
+5) The `printAddedModule` method is called from the `Print` class to display the result to the user.
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![addInputCommand](diagrams/AddInputCommand.png)
+
+<small><i>Figure ???</i></small>
+
+### 3.4. Find Modules
+
+The find modules features is facilitated by the `Parser`, `ModuleList` and `Module` Classes. `Parser` helps to check
+for user input to determine is user doing a search via a module name or module type, which will then invoke the 
+`findModuleByName` and `findModuleByType` methods respectively. If any modules are found, the ChatBot will print out
+all the modules, else the ChatBot will return a message to the user, saying that no modules were found with the search
+term they used.
+
+Given below is an example usage scenario and how the findModuleByName mechanism behaves at each step.
+
+Step 1: The user launches the application for the first time. The ChatBot will be initialised with an empty ArrayList
+of Modules.
+
+Step 2: The user executes `add /CS2113T CS2101 /4 /CORE /2 /2` to add two modules into the ChatBot via the addModules 
+feature
+
+The following object diagram shows the current state of the ChatBot:
+![FindModule](diagrams/FindTwoModules.png)
+
+<small><i>Figure ???</i></small>
+
+Step 3: The user now wants to find the details of the modules he has just added. He executes `find /name /CS` which
+will call `findModuleByName`. This method will then print out all the modules that have the keyword "CS" in their
+module name.
+
+The following sequence diagram shows how the `findModuleByName` operation works:
+
+![FindModule](diagrams/FindModule.png)
```

Comment:
Bar ends before the return of self invocation method
<img width="512" alt="image" src="https://user-images.githubusercontent.com/88694646/228722901-675d1502-31d6-4f3a-9a98-50bc8bddb016.png">